File: src/priv/bajdcc/syntax/automata/npa/NPAInstruction.java
package priv.bajdcc.syntax.automata.npa;

/**
 * 非确定性下推自动机指令
 *
 * @author bajdcc
 */
public enum NPAInstruction {
	PASS("通过"), READ("读入"), SHIFT("移进"), TRANSLATE("归约"), LEFT_RECURSION(
			"左递归"), TRANSLATE_DISCARD("归约"), LEFT_RECURSION_DISCARD("左递归"), TRANSLATE_FINISH(
			"归约结束");

	private String name;

	NPAInstruction(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
}

File: src/main/java/com/bajdcc/LALR1/syntax/automata/npa/NPAInstruction.kt
package com.bajdcc.LALR1.syntax.automata.npa

/**
 * 非确定性下推自动机指令
 *
 * @author bajdcc
 */
enum class NPAInstruction constructor(var desc: String?) {
    PASS("通过"),
    READ("读入"),
    SHIFT("移进"),
    TRANSLATE("翻译"),
    LEFT_RECURSION("左递归"),
    TRANSLATE_DISCARD("翻译"),
    LEFT_RECURSION_DISCARD("左递归"),
    TRANSLATE_FINISH("翻译结束")
}
